kartothek/core/dataset.py

```diff
@@ -234,6 +223,42 @@ def to_json(self) -> bytes:
     def to_msgpack(self) -> bytes:
         return packb(self.to_dict())
 
+    def load_partition_indices(self: T) -> T:
+        """
+        Load all filename encoded indices into RAM. File encoded indices can be extracted from datasets with partitions
+        stored in a format like
+
+        .. code::
+
+            `dataset_uuid/table/IndexCol=IndexValue/SecondIndexCol=Value/partition_label.parquet`
+
+        Which results in an in-memory index holding the information
+
+        .. code::
+
+            {
+                "IndexCol": {
+                    IndexValue: ["partition_label"]
+                },
+                "SecondIndexCol": {
+                    Value: ["partition_label"]
+                }
+            }
+
+        """
+        if self.primary_indices_loaded:
+            return self
+
+        indices = _construct_dynamic_index_from_partitions(
+            partitions=self.partitions,
+            schema=self.schema,
+            default_dtype=pa.string() if self.metadata_version == 3 else None,
+            partition_keys=self.partition_keys,
+        )
+        combined_indices = self.indices.copy()
+        combined_indices.update(indices)
+        return self.copy(indices=combined_indices)
```

Comment:
Maybe I do miss it , but as I currently understand if `self.primary_indices_loaded` is True we return the object itself. If not we load the indices and return a copy. So `self.primary_indices_loaded` is not updated from what I see, so I wonder why not?
Why even the copy?

Reply:
The `copy` methods were introduced at the very beginning of `kartothek` and we perceived it as a "clean coding" by not mutating any objects inplace but always return a new instance. This is what the `copy` methods are doing. By now I consider this unnecessary boilerplate but I didn't want to break any patterns here. Therefore if nothing is to be done, the same instance is returned, otherwise a new instance with the mutated state is returned

FYI this method is unchanged but was moved for some reason. I'll move it back to have cleaner diffs / blames. still valid question, of course :)

Reply:
Ah okay got it. I understand the arguments of preventing mutation but also that not everything can and should be folloewed 100%  and here I might argue against it but as it is pre-change code I am totally fine :)